explain-walkthrough.md

```diff
@@ -0,0 +1,206 @@
+
+{{< copyable "sql" >}}
+
+```sql
+EXPLAIN SELECT count(*) FROM trips WHERE start_date BETWEEN '2017-07-01 00:00:00' AND '2017-07-01 23:59:59';
+```
+
+```sql
++------------------------------+----------+-----------+---------------+------------------------------------------------------------------------------------------------------------------------+
+| id                           | estRows  | task      | access object | operator info                                                                                                          |
++------------------------------+----------+-----------+---------------+------------------------------------------------------------------------------------------------------------------------+
+| StreamAgg_20                 | 1.00     | root      |               | funcs:count(Column#13)->Column#11                                                                                      |
+| └─TableReader_21             | 1.00     | root      |               | data:StreamAgg_9                                                                                                       |
+|   └─StreamAgg_9              | 1.00     | cop[tikv] |               | funcs:count(1)->Column#13                                                                                              |
+|     └─Selection_19           | 250.00   | cop[tikv] |               | ge(bikeshare.trips.start_date, 2017-07-01 00:00:00.000000), le(bikeshare.trips.start_date, 2017-07-01 23:59:59.000000) |
+|       └─TableFullScan_18     | 10000.00 | cop[tikv] | table:trips   | keep order:false, stats:pseudo                                                                                         |
++------------------------------+----------+-----------+---------------+------------------------------------------------------------------------------------------------------------------------+
+5 rows in set (0.00 sec)
+```
+
+Working from the child operator `└─TableFullScan_18` back, we can see that this query is currently suboptimal:
+
+1. The coprocessor (TiKV) is reading the entire trips table, as a `TableFullScan`. It then passes the rows that it reads to `Selection_19`, still within TiKV.
+2. The `WHERE start_date BETWEEN ..` predicate is then filtered in the `Selection_19` operator. It is estimated that approximately `250` rows will meet this selection. Note that this number was produced from a heuristic; the `└─TableFullScan_18` operator shows `stats:pseudo`. After running `ANALYZE TABLE trips` the statistics should be more accurate.
+3. The rows that met the selection criteria then have a `count` function applied to them. This is also completed inside the `StreamAgg_9` operator, which is still inside TiKV (`cop[tikv]`). The TiKV coprocessor understands a number of MySQL built-in functions, `count` being one of them. It also understands that `count` is safe to apply Stream Aggregation to, even though the results are not in order.
```

Comment:
I can link to aggregation once https://github.com/pingcap/docs/pull/3926 merges. Stream aggregation uses less memory than hashagg, but has rules for if it can be used.